### PR TITLE
fix:sherlock

### PR DESCRIPTION
**Pull request by Gevarist**

### CODE REVIEW

File: src/ArrakisPublicVaultRouter.sol [hunk 1]
```diff
@@ -1225,8 +1225,10 @@ contract ArrakisPublicVaultRouter is
         uint256 proportion =
             proportion0 < proportion1 ? proportion0 : proportion1;
 
-        amount0ToDeposit = FullMath.mulDiv(amount0, proportion, BASE);
-        amount1ToDeposit = FullMath.mulDiv(amount1, proportion, BASE);
+        amount0ToDeposit =
```
Comment (CergyK) [inline src/ArrakisPublicVaultRouter.sol:1228]: fix ok for https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/54

File: src/ArrakisStandardManager.sol [hunk 2]
```diff
@@ -413,6 +430,15 @@ contract ArrakisStandardManager is
             }
         }
 
+        /// @dev only public vault should have this check.
+        if (isPublicVault) {
```
Comment (CergyK) [inline src/ArrakisStandardManager.sol:434]: fix ok for https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/44

Comment (CergyK) [reply src/ArrakisStandardManager.sol:434]: and https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/76

File: src/abstracts/ArrakisMetaVault.sol [hunk 3]
```diff
@@ -126,9 +126,32 @@ abstract contract ArrakisMetaVault is
...
+        }
+
+        _call(module_, payloads_[0]);
+
+        // #endregion first call.
+
+        for (uint256 i = 1; i < len; i++) {
+            selector = bytes4(payloads_[i][:4]);
+
+            if (IArrakisLPModule.withdraw.selector == selector) {
```
Comment (CergyK) [inline src/abstracts/ArrakisMetaVault.sol:150]: fix ok for https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/50

File: src/modules/ValantisHOTModulePublic.sol [hunk 4]
```diff
@@ -107,4 +107,13 @@ contract ValantisModulePublic is
         }
         return super.withdraw(receiver_, proportion_);
     }
+
+    function initializePosition(bytes calldata data_)
+        external
+        override
+        onlyMetaVault
+    {
+        notFirstDeposit = true;
```
Comment (CergyK) [inline src/modules/ValantisHOTModulePublic.sol:116]: fix ok for https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/27

File: src/RouterSwapExecutor.sol [hunk 5]
```diff
@@ -54,7 +54,7 @@ contract RouterSwapExecutor is IRouterSwapExecutor {
             if (token0 != nativeToken) {
                 balanceBefore =
                     IERC20(token0).balanceOf(address(this));
-                IERC20(token0).safeIncreaseAllowance(
```
Comment (CergyK) [inline src/RouterSwapExecutor.sol:57]: fix ok for https://github.com/sherlock-audit/2024-03-arrakis-judging/issues/19

**Review by kassandraoftroy — APPROVED** [review]
LGTM

**Pull request merged**